.github/workflows/stage_release.yml

```diff
@@ -0,0 +1,126 @@
+
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: 'Set up Java'
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          server-id: ossrh
+          server-username: SONAR_TYPE_TOKEN_USERNAME # env variable for Sonar Type username
+          server-password: SONAR_TYPE_TOKEN_PASSWORD # env variable for Sonar Type password
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE # env variable for GPG private key passphrase
+          gpg-private-key: ${{ secrets.KROXYLICIOUS_RELEASE_GPG_PRIVATE_KEY }} # Value of the GPG private key to import
+          overwrite-settings: true
+
+      - name: 'Verify no existing staging repository exists'
+        if: ${{ github.event.inputs.dry-run == 'false' }}
+        env:
+          SONAR_TYPE_TOKEN_USERNAME: ${{ vars.KROXYLICIOUS_SONAR_TYPE_TOKEN_USERNAME }}
+          SONAR_TYPE_TOKEN_PASSWORD: ${{ secrets.KROXYLICIOUS_SONAR_TYPE_TOKEN_PASSWORD }}
+        run: ./scripts/transition-staging-repository-state.sh -a
+
+      - name: 'Stage Release'
+        run: |
+          SIGNING_KEY_SHORT_NAME=$(gpg --list-public-keys --keyid-format short --with-colons | awk -F: '/^pub:/ {print $5}')
+          ./scripts/stage-release.sh -k ${SIGNING_KEY_SHORT_NAME} \
+                                     -v ${{ github.event.inputs.release-version }} \
+                                     -b ${{ github.event.inputs.branch }} \
+                                     ${{ github.event.inputs.dry-run == 'true' && '-d' || '' }} \
+                                     ${{ github.event.inputs.skip-tests == 'true' && '-s' || '' }}
+        env:
+          SONAR_TYPE_TOKEN_USERNAME: ${{ vars.KROXYLICIOUS_SONAR_TYPE_TOKEN_USERNAME }}
+          SONAR_TYPE_TOKEN_PASSWORD: ${{ secrets.KROXYLICIOUS_SONAR_TYPE_TOKEN_PASSWORD }}
+          MAVEN_GPG_PASSPHRASE: ${{ secrets.KROXYLICIOUS_RELEASE_GPG_PASSPHRASE }}
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }} # For the gh cmd line tool used by stage-release.sh
+
+      - name: 'Drop stage repository, if failed'
+        if: ${{ failure() && github.event.inputs.dry-run == 'false' }}
+        env:
+          SONAR_TYPE_TOKEN_USERNAME: ${{ vars.KROXYLICIOUS_SONAR_TYPE_TOKEN_USERNAME }}
+          SONAR_TYPE_TOKEN_PASSWORD: ${{ secrets.KROXYLICIOUS_SONAR_TYPE_TOKEN_PASSWORD }}
+        run: ./scripts/transition-staging-repository-state.sh -s drop
+
+      - name: 'Close stage repository'
+        if: ${{ github.event.inputs.dry-run == 'false' }}
+        env:
+          SONAR_TYPE_TOKEN_USERNAME: ${{ vars.KROXYLICIOUS_SONAR_TYPE_TOKEN_USERNAME }}
+          SONAR_TYPE_TOKEN_PASSWORD: ${{ secrets.KROXYLICIOUS_SONAR_TYPE_TOKEN_PASSWORD }}
+        run: ./scripts/transition-staging-repository-state.sh -s close
```

Comment:
So, we could omit this `close` step if we switched the pom.xml to use `org.sonatype.plugins:nexus-staging-maven-plugin` rather than the neutral `maven-deploy-plugin`.    We'd then configure nexus-staging-maven-plugin to [skipStagingRepositoryClose=false](https://github.com/sonatype/nexus-maven-plugins/blob/main/staging/maven-plugin/README.md#:~:text=staging%20repository%20(boolean)-,skipStagingRepositoryClose,-false).   I think we'd still want the verification step, so we'd still need the transition-staging-repository-state.sh to handle the other state transitions.

The disadvantage of using nexus-staging-maven-plugin in the POM is that we then tie the project to Nexus.

Reply:
We are tied to Nexus as that is what runs Maven central, that said I think we should stick to the deploy plugin.